sync-core/src/test/java/com/cloudant/sync/util/TestUtils.java

```diff
@@ -153,5 +148,27 @@ public static File loadFixture(String fileName){
        }
    }
 
+   public static SQLDatabase getDatabaseConnectionToExsitingDb(SQLDatabase db){
+       if(Misc.isRunningOnAndroid())
+           return db;
+
+       try {
+           String filePath = (String) db.getClass()
```

Comment:
Yeah, I think it was because originally, when I removed the threadlocal code, the tests didn't have access to the internal queues, looking at it again, I should change the tests to run db actions on the internal queues.